fsck/fsck.c

```diff
@@ -1275,6 +1275,57 @@ static int read_upcase_table(struct exfat_fsck *fsck)
 	return retval;
 }
 
+/*
+ * Checks whether there are other directory entries following the unused
+ * directory entries. If so, sets the unused directory entries to the deleted
+ * directory entries(Type 0x7F).
+ */
+static int check_unused_dentry(struct exfat_de_iter *de_iter,
+		struct exfat_inode *dir)
+{
+	int ret, i;
+	struct exfat_dentry *dentry;
+	int num_unused = 1;
+
+	while (1) {
+		exfat_de_iter_advance(de_iter, 1);
+		ret = exfat_de_iter_get(de_iter, 0, &dentry);
+		if (ret == EOF)
+			return 0;
+		else if (ret) {
+			fsck_err(dir->parent, dir,
+				"failed to get a dentry. %d\n", ret);
+			return ret;
+		}
+
+		if (dentry->type != EXFAT_LAST)
+			break;
+
+		num_unused++;
+	}
+
+	if (!repair_file_ask(de_iter, NULL, ER_DE_UNUSED,
+			    "other entry(type: 0x%02X) follows unused entry",
+			    dentry->type))
+		return -EINVAL;
+
+	ret = exfat_de_iter_revert(de_iter, num_unused);
+	if (ret < 0) {
+		fsck_err(dir->parent, dir,
+			"failed to revert %d dentries. %d\n", num_unused, ret);
+		return ret;
+	}
+
+	for (i = 0; i < num_unused; i++) {
+		exfat_de_iter_get_dirty(de_iter, 0, &dentry);
+		dentry->type = 0x7F;
```

Comment:
This patch looks good to me, but  Could you replace this magic value with some macro?

Reply:
@YuezhangMo He may not have checked this review point.